client/me/help/help-contact-form/index.jsx

```diff
@@ -109,6 +111,7 @@ export const HelpContactForm = React.createClass( {
 	setSite( siteSlug ) {
 		const site = sites.getSite( siteSlug );
 		this.setState( { siteId: site.ID } );
+		this.props.onChangeSite( site.ID );
```

Comment:
I totally agree, but when I worked on this I had to use QuerySites because `sites-list` was not fully removed from Calypso yet, so I had to hack it using component state. That is not the case anymore and I switched to full redux.